main.go

```diff
@@ -0,0 +1,153 @@
+package main
+
+import (
+	"abuse-scanner/database"
+	"abuse-scanner/email"
+	"errors"
+	"fmt"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"context"
+	"log"
+	"os"
+
+	"github.com/joho/godotenv"
+	"github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func main() {
+	// load env
+	_ = godotenv.Load()
+
+	// create a context
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Contexts are great for controlling blocking/ongoing operations but for things like a scanner, server etc. `Close` or `Shutdown` methods are nicer. 

Another option would be to, instead of using a waitgroup, pass the context to the individual `Start` methods. But I'd either go for.

1. waitgroup within the objects with a `Close`/`Shutdown` and `Start()`
2. context that gets passed to `Start(ctx)` but no waitgroup.

Reply:
I used a waitgroup and added `Start` and `Stop` methods with a timeout on the `Wait` in the waitgroup.